node/src/blockchain/blockchain_interface.rs

```diff
@@ -251,13 +275,30 @@ where
             )
             .build();
 
-        let log_request = self.web3.eth().logs(filter);
+        let fallback_end_block_number = match end_block {
+            BlockNumber::Number(eb) => eb.as_u64(),
+            _ => {
+                if let BlockNumber::Number(start_block_number) = start_block {
+                    start_block_number.as_u64() + 1u64
+                } else {
+                    panic!("start_block of Latest, Earliest, and Pending are not supported");
+                }
+            }
+        };
+        let block_request = self.web3_batch.eth().block_number();
+        let log_request = self.web3_batch.eth().logs(filter);
+
         let logger = self.logger.clone();
-        log_request
-            .then(|logs| {
-                debug!(logger, "Transaction retrieval completed: {:?}", logs);
-                future::result::<RetrievedBlockchainTransactions, BlockchainError>(match logs {
+        match self.web3_batch.transport().submit_batch().wait() {
+            Ok(_) => {
+                let response_block_number = match block_request.wait() {
+                    Ok(block_nbr) => block_nbr.as_u64(),
```

Comment:
Also here, this would be such good information for a debugging person. Please add the debug log saying what current head block was reported from the blockchain.